attribute/reference.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2017 Istio Authors
```

Comment:
If this stuff relevant? It seems like more wire-protocol stuff, right?

Reply:
Runtime stringmaps need to callback to the bag to record accesses. Reference tracking is optional in this library, but it is required in istio for now, until we can delete the wire stuff.